lib/utils/ICETelescope.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef ICE_TELESCOPE_HPP
+#define ICE_TELESCOPE_HPP
+
+#include "Config.hpp" // for Config
+#include "Telescope.hpp"
+#include "chimeMetadata.h"
+
+#include <stdint.h> // for int32_t, uint32_t
+#include <string>   // for string
+#include <time.h>
+
+/**
+ * @brief Implementation for an ICEboard like telescope with a Casper PFB.
+ *
+ * @conf  sample_rate   The sampling rate in MHz.
+ * @conf  frame_length  The FPGA frame length in samples.
+ * @conf  nyquist_zone  The Nyquist zone we are sampling in (zone=1 is standard
+ *                      sampling, zone>=2 is alias sampling).
+ **/
+class ICETelescope : public Telescope {
+public:
+    ICETelescope(const kotekan::Config& config, const std::string& path);
+
+    // Implementations of the required frequency mapping functions
+    freq_id_t to_freq_id(stream_t stream, uint32_t ind) const override;
+    double to_freq(freq_id_t freq_id) const override;
+    uint32_t num_freq_per_stream() const override;
+
+    // Implementations of the required time mapping functions
+    timespec to_time(uint64_t seq) const override;
+    uint64_t to_seq(timespec time) const override;
+
+protected:
+    /**
+     * @brief Construct an ICEboard style telescope
+     *
+     * @param  sample_rate  The sampling rate in MHz.
+     * @param  length       The length of each frame.
+     * @param  zone         Which Nyquist zone are we sampling in? zone=1 is
+     *                      standard sampling, zone>=2 are alias sampling.
+     **/
+    ICETelescope(double sample_rate, uint32_t length, uint8_t zone, const kotekan::Config& config,
+                 const std::string& path);
+
+private:
```

Comment:
Can we make these `protected` so the subclasses can access them?

Reply:
Fine by me. I was just going to push a bunch of changes. Let me add that one and push in a minute.